packages/flutter/lib/src/material/user_accounts_drawer_header.dart

```diff
@@ -60,75 +62,119 @@ class _AccountPictures extends StatelessWidget {
   }
 }
 
-class _AccountDetails extends StatelessWidget {
+class _AccountDetails extends StatefulWidget {
   const _AccountDetails({
-    Key key,
     @required this.accountName,
     @required this.accountEmail,
     this.onTap,
     this.isOpen,
-  }) : super(key: key);
+  });
 
   final Widget accountName;
   final Widget accountEmail;
   final VoidCallback onTap;
   final bool isOpen;
 
+  @override
+  _AccountDetailsState createState() => new _AccountDetailsState();
+}
+
+class _AccountDetailsState extends State<_AccountDetails> with SingleTickerProviderStateMixin {
+  Animation<double> _animation;
+  AnimationController _controller;
+
+  @override
+  void initState () {
+    super.initState();
+
+    _controller = new AnimationController(
+      value: widget.isOpen ? 1.0 : 0.0,
+      duration: const Duration(milliseconds: 200),
+      vsync: this,
+    );
+    _animation = new CurvedAnimation(
+        parent: _controller,
+        curve: Curves.easeInOut,
+      )
+      ..addListener(() {
+        setState(() {});
+      });
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void _onTap() {
+    if (widget.isOpen) {
+      _controller.reverse();
```

Comment:
this doesn't seem right. Consider the case where the user taps the widget, onTap is invoked, and does _not_ change isOpen. Now the arrow will be animating the wrong direction.

I think it would make more sense to trigger the animation during didUpdateWidget based on the new value of isOpen.

See `data_table.dart` in the material directory, there's a _SortArrow widget there that you may wish to crib from. (It also does other cool things, like always going clockwise unless you change it mid-turn, IIRC.)